commands/command_pre_push.go

```diff
@@ -137,7 +133,19 @@ func prePushRef(left, right string) {
 			os.Exit(2)
 		}
 	}
+}
+
+func filteredPointers(pointers []*lfs.WrappedPointer, filter lfs.StringSet) []*lfs.WrappedPointer {
```

Comment:
> introduces a concurrency issue because the second filter call is looping through the same slice while the callback is appending to it

I don't think it does - the filteredPointers and prePushCheckForMissingObjects calls are completely synchronous/serial. But in any case, if you're happier with this style it's fine, I just can't help myself looking for allocation optimisations given my background ;)